confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java

```diff
@@ -204,17 +204,24 @@ public int getTotalCpuCoreCount() {
    * Only leader use this interface
    *
    * @param dataNodeId Specific DataNodeId
-   * @return All registered DataNodes if dataNodeId equals -1. And return the specific DataNode
-   *     otherwise.
+   * @return All registered DataNodes
    */
-  public List<TDataNodeConfiguration> getRegisteredDataNodes(int dataNodeId) {
-    return nodeInfo.getRegisteredDataNodes(dataNodeId);
+  public List<TDataNodeConfiguration> getRegisteredDataNodes() {
+    return nodeInfo.getRegisteredDataNodes();
   }
 
-  public Map<Integer, TDataNodeLocation> getRegisteredDataNodeLocations(int dataNodeId) {
+  public Map<Integer, TDataNodeLocation> getRegisteredDataNodeLocationById(int dataNodeId) {
+    Map<Integer, TDataNodeLocation> dataNodeLocations = new ConcurrentHashMap<>();
+    dataNodeLocations.put(
+        nodeInfo.getRegisteredDataNodeById(dataNodeId).getLocation().getDataNodeId(),
+        nodeInfo.getRegisteredDataNodeById(dataNodeId).getLocation());
+    return dataNodeLocations;
+  }
```

Comment:
An interface that returns a map with only one element looks a little strange?

Reply:
> An interface that returns a map with only one element looks a little strange?
Okay, I've modified it ~